pyproject.toml

```diff
@@ -51,7 +51,6 @@ requests = "*"
 cryptography = "*"
 packaging = "*"
 "ruamel.yaml" = "*"
-pathspec = "*"
 rich = "*"
```

Comment:
I'm just curious because it's the first time I noticed this, but is it normal with poetry to specify `*` as dependency version?

Wouldn't this be prone to breakage with `poetry update` (or whatever poetry's update command is).

Reply:
Specifying dependencies like this is not the default in `poetry` but it is preferred so as _not_ to use upper bound version constraints. See [this blog post for the source of that decision for this repository](https://iscinumpy.dev/post/bound-version-constraints/) and the [associated guidance in `CONTRIBUTING.md`](https://github.com/phylum-dev/phylum-ci/blob/main/CONTRIBUTING.md?plain=1#L160-L190).

Dependencies will be checked automatically in CI during a PR (required for all proposed changes) with Phylum and the test suite. There is also an automated weekly dependency bump. For each of these PRs, new dependencies and major version updates are further inspected (review changelog entries, etc.) to help to identify and prevent breakage and/or take advantage of new features. This process has worked well so far, with the last example of a major version breaking change being captured [with this issue](https://github.com/phylum-dev/phylum-ci/issues/278) (and a pin in the manifest).